javagony/WhileLoop.java

```diff
@@ -0,0 +1,49 @@
+import java.util.function.BooleanSupplier;
+
+public class WhileLoop {
+	/**
+	 * Executes the body in a loop while the condition is true, checking the condition at the
+	 * beginning of each loop iteration.
+	 * 
+	 * @param condition the condition to check
+	 * @param body      the body of the while loop
+	 */
+	public static void whileLoop(BooleanSupplier condition, Runnable body) {
+		try {
+			@SuppressWarnings("unused")

```

Comment:
I never thought about using this decorator. I should probably use it in the other examples 😆